src/Microsoft.AspNetCore.Blazor/RenderTree/RenderTreeBuilder.cs

```diff
@@ -97,45 +97,90 @@ public void AddContent(int sequence, RenderFragment fragment)
         public void AddContent(int sequence, object textContent)
             => AddContent(sequence, textContent?.ToString());
 
+        /// <summary>
+        /// Appends a frame representing a bool-valued attribute.
+        /// The attribute is associated with the most recently added element. If the value is <c>false</c> and the
+        /// current element is not a component, the frame will be omitted.
+        /// </summary>
+        /// <param name="sequence">An integer that represents the position of the instruction in the source code.</param>
+        /// <param name="name">The name of the attribute.</param>
+        /// <param name="value">The value of the attribute.</param>
+        public void AddAttribute(int sequence, string name, bool value)
+        {
+            AssertCanAddAttribute();
+            if (value || _lastNonAttributeFrameType == RenderTreeFrameType.Component)
+            {
+                Append(RenderTreeFrame.Attribute(sequence, name, true));
```

Comment:
Is this going to create a separate boxed `bool` value on every invocation? If so, would we be able to replace the `true` literal here with something like `"true"` so it's the same reference-type instance on every invocation?

Reply:
Fixed this by using cached statics of the boxed values.

This removes the boxing as well for components, since we'd always box in those cases.